inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp

```diff
@@ -1648,273 +1629,302 @@ SizeVector to5Dim(SizeVector casesDim) {
+            return false;
+        }
+        const auto &interpAttr = interp->get_attrs();
+
+        const size_t dataRank = interp->get_input_shape(DATA_ID).size();
+        if (dataRank < 1 || dataRank > 5)
+            THROW_IE_EXCEPTION_WITH_STATUS(NOT_IMPLEMENTED) << "Does not support input tensor of rank : " << dataRank;
+        if (dataRank == 5 && interpAttr.mode == ngraph::opset4::Interpolate::InterpolateMode::cubic)
+            THROW_IE_EXCEPTION_WITH_STATUS(NOT_IMPLEMENTED) << "Doesn't support input tensor with rank: " << dataRank << " for 'cubic' mode ";
+
+        if (std::dynamic_pointer_cast<const ngraph::opset1::Constant>(interp->get_input_node_shared_ptr(SCALES_ID)) == nullptr)
+            THROW_IE_EXCEPTION_WITH_STATUS(NOT_IMPLEMENTED) << "Only const 'scales' input is supported";
+        if (interp->get_input_size() > 3 && std::dynamic_pointer_cast<const ngraph::opset1::Constant>(interp->get_input_node_shared_ptr(AXES_ID)) == nullptr)
+            THROW_IE_EXCEPTION_WITH_STATUS(NOT_IMPLEMENTED) << "Only const 'axes' input is supported";
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+MKLDNNInterpolateNode::MKLDNNInterpolateNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache)
+        : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    if (isSupportedOperation(op, errorMessage)) {
+        errorPrefix = "Interpolate node with name '" + getName() + "'";
+
+        const auto interp = std::dynamic_pointer_cast<const ngraph::opset4::Interpolate>(op);
+
+        if (interp->get_input_size() != 3 && interp->get_input_size() != 4)
+            THROW_IE_EXCEPTION << errorPrefix << " has incorrect number of input edges";
+        if (interp->get_output_size() != 1)
+            THROW_IE_EXCEPTION << errorPrefix << " has incorrect number of output edges";
+        isAxesSpecified = interp->get_input_size() != 3;
+
+        const auto &interpAttr = interp->get_attrs();
+
+        const size_t dataRank = interp->get_input_shape(DATA_ID).size();
+        const auto &interpMode = interpAttr.mode;
+        if (interpMode == ngraph::opset4::Interpolate::InterpolateMode::nearest) {
+            mode = InterpolateMode::nearest;
+        } else if (interpMode == ngraph::opset4::Interpolate::InterpolateMode::linear) {
+            if (dataRank < 5) {
+                mode = InterpolateMode::linear_onnx;
+            } else {
+                mode = InterpolateMode::linear;
+            }
+        } else if (interpMode == ngraph::opset4::Interpolate::InterpolateMode::linear_onnx) {
+            mode = InterpolateMode::linear_onnx;
+        } else if (interpMode == ngraph::opset4::Interpolate::InterpolateMode::cubic) {
+            mode = InterpolateMode::cubic;
+        } else {
+            THROW_IE_EXCEPTION << errorPrefix << " does not support interpolate mode: " << ngraph::as_string(interpMode);
```

Comment:
This should be verified inside `isSupportedOperation`. The same for: interpCoordTransMode, interpNearestMode, interpShapeCalcMode.

Reply:
moved into ```isSupportedOperation```